test/e2e/syncer/syncer_test.go

```diff
@@ -229,6 +247,69 @@ func TestSyncerLifecycle(t *testing.T) {
 		return true
 	}, wait.ForeverTestTimeout, time.Millisecond*100, "downstream deployment %s/%s was not synced", downstreamNamespaceName, upstreamDeployment.Name)
 
+	// Add a virtual Finalizer to the deployment and update it.
+	t.Logf("Adding a virtual finalizer to the upstream deployment %s/%s in order to simulate an external controller", upstreamNamespace.Name, upstreamDeployment.Name)
+	upstreamDeployment, err = upstreamKubeClient.AppsV1().Deployments(upstreamNamespace.Name).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	upstreamDeployment = upstreamDeployment.DeepCopy()
+	annotations := upstreamDeployment.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string, 0)
+	}
+	annotations["finalizers.workload.kcp.dev/"+syncerFixture.SyncerConfig.WorkloadClusterName] = "external-controller-finalizer"
+	upstreamDeployment.SetAnnotations(annotations)
+	_, err = upstreamKubeClient.AppsV1().Deployments(upstreamNamespace.Name).Update(ctx, upstreamDeployment, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Deleting upstream deployment %s/%s", upstreamNamespace.Name, upstreamDeployment.Name)
+	err = upstreamKubeClient.AppsV1().Deployments(upstreamNamespace.Name).Delete(ctx, upstreamDeployment.Name, metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64(0)})
+	require.NoError(t, err)
+
+	t.Logf("Checking if upstream deployment %s/%s is getting deleted, shouldn't as there is a virtual finalizer", upstreamNamespace.Name, upstreamDeployment.Name)
+	require.Never(t, func() bool {
+		if _, err := upstreamKubeClient.AppsV1().Deployments(upstreamNamespace.Name).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				t.Errorf("Deployment %s/%s was deleted", upstreamNamespace.Name, upstreamDeployment.Name)
+			} else {
+				t.Errorf("saw an error trying to get the upstream deployment %s/%s, err: %v", upstreamNamespace.Name, upstreamDeployment.Name, err)
+			}
+			return true
+		}
+		return false
+	}, 5*time.Second, time.Second, "upstream Deployment got deleted or there was an error", upstreamNamespace.Name, upstreamDeployment.Name)
+
+	t.Logf("Checking if the downstream deployment %s/%s is deleted or not (shouldn't as there's a virtual finalizer)", upstreamNamespace.Name, upstreamDeployment.Name)
```

Comment:
Should we test that the deletion annotation timestamp has been added to the upstream deployment here (with Eventually) ?

Reply:
done